pkg/pipeline/common_steps.go

```diff
@@ -69,29 +74,49 @@ func addTenantLabel(obj PipelineObject, data *ExecutionContext) ExecutionResult
 
 func updateObject(obj PipelineObject, data *ExecutionContext) ExecutionResult {
 
-	oldResourceVersion := obj.GetObjectMeta().GetResourceVersion()
-
 	rtObj, ok := obj.(runtime.Object)
 	if !ok {
-		return ExecutionResult{
-			Abort: true,
-			Err:   fmt.Errorf("object ist not a valid runtime.object: %v", obj.GetObjectMeta().GetName()),
+		return ExecutionResult{Err: errors.New("object is not a runtime object")}
+	}
+
+	if !specAndMetaEqual(obj, data.originalObject) {
+		err := data.Client.Update(context.TODO(), rtObj)
+		if err != nil {
+			if k8serrors.IsConflict(err) {
+				return ExecutionResult{Abort: true}
+			}
+			return ExecutionResult{Err: err}
 		}
 	}
 
-	updatedObj := rtObj.DeepCopyObject()
-	err := data.Client.Update(context.TODO(), updatedObj)
-	if err != nil {
-		return ExecutionResult{Err: err}
+	if !equality.Semantic.DeepEqual(data.originalObject.GetStatus(), obj.GetStatus()) {
+
+		origRtObj := data.originalObject.(runtime.Object)
+
+		err := data.Client.Status().Patch(context.TODO(), rtObj, client.MergeFrom(origRtObj))
+		if err != nil {
+			if k8serrors.IsConflict(err) {
```

Comment:
Apparently for `Patch` requests, the concurrency mechanism is skipped, so I don't see how we could end up with a conflict error here anyway.